### DISCUSSION TITLE
Can the JoinHandle returned by task::spawn contain an error if the async function is infallible?

### DISCUSSION THREAD

**Discussion by nvartolomei** [tokio-rs/tokio#5744] [Q&A]
The question is about whether we should always await the JoinHandle or it can be omitted in some circumstances. E.g. I can imagine a case where a OS thread is killed, is that a valid concern?

**Comment by hawkw** [comment]
A `JoinHandle` future may complete with an error if the spawned task is cancelled (via `JoinHandle::abort`) or if the task panics.

**Reply by Darksonn** [reply]
A task can also be cancelled if the runtime is shut down.
